framework/src/modules/chain/logic/process_transaction.js

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright © 2018 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ */
+
+'use strict';
+
+const { Status: TransactionStatus } = require('@liskhq/lisk-transactions');
+
+/**
+ * Executes each step from left to right and pipes the transactions that succeed to the next
+ * step. Finally collects all responses and formats them accordingly.
+ * @param steps
+ * @returns {function(*=): {transactionsResponses: *[]}}
+ */
+const composeProcessTransactionSteps = (...steps) => async transactions => {
```

Comment:
This name does not match with its usage. You are not using it just to compose the process transactions steps, but verify steps as well. So better may be just `composeTransactionSteps`